src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java

```diff
@@ -119,29 +135,40 @@ public void messageOnServerThread(final ChangeFreeToInteractBlockMessage message
             //Verify player has permission to change this huts settings
             if (!colony.getPermissions().hasPermission(player, Permissions.Action.EDIT_PERMISSIONS))
             {
+                LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.permission.deny");
                 return;
             }
 
             if(message.type == MessageType.ADD_BLOCK)
             {
-                if (!(message.pos.getX() == 0 && message.pos.getZ() == 0 && message.pos.getY() == 0))
-                {
-                    colony.addFreeBlock(message.block);
-                }
-                else
+                switch (message.mode)
                 {
-                    colony.addFreePosition(message.pos);
+                    case LOCATION:
+                        colony.addFreePosition(message.pos);
+                        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.addposition.success", message.pos.getX(),message.pos.getY(),message.pos.getZ());
+                        break;
+                    case BLOCK:
+                        colony.addFreeBlock(message.block);
+                        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.addblock.success", message.block.getRegistryName());
+                        break;
+                    default:
+                        // Error!
                 }
             }
             else
             {
-                if (!(message.pos.getX() == 0 && message.pos.getZ() == 0 && message.pos.getY() == 0))
-                {
-                    colony.removeFreeBlock(message.block);
-                }
-                else
+                switch (message.mode)
                 {
-                    colony.removeFreePosition(message.pos);
+                    case LOCATION:
+                        colony.removeFreePosition(message.pos);
+                        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.removelocation.success", message.pos.getX(),message.pos.getY(),message.pos.getZ());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 198 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)